omnipaxos_storage/src/memory.rs

```diff
@@ -0,0 +1,374 @@
+                }
+                entries.push(FromBytes::read_from(msg.payload()).unwrap());
+            }
+            println!("res from get_entries {:?}", entries);
+            entries
+        }
+
+        fn get_log_len(&self) -> u64 {
+            self.c_log.next_offset()
+            //println!("log length: {:?}", res);
+            //res
+        }
+
+        fn get_suffix(&self, from: u64) -> Vec<T> {
+            //println!("get suffix!");
+            self.get_entries(from, self.get_log_len())
+        }
+
+        fn get_promise(&self) -> Ballot {
+            match self.db.get(b"n_prom") {
+                Ok(Some(mut value)) => {
+                    let prom_bytes: &mut [u8] = &mut value;
+                    FromBytes::read_from(prom_bytes).unwrap()
+                }
+                Ok(None) => Ballot::default(),
+                Err(_e) => Ballot::default(),
+            }
+        }
+
+        fn set_promise(&mut self, n_prom: Ballot) {
+            let prom_bytes = AsBytes::as_bytes(&n_prom);
+            self.db.put(b"n_prom", prom_bytes).unwrap()
+        }
+
+        fn get_decided_idx(&self) -> u64 {
+            match self.db.get(b"ld") {
+                Ok(Some(value)) => {
+                    let ld_bytes: &[u8] = &value;
+                    FromBytes::read_from(ld_bytes).unwrap()
+                }
+                Ok(None) => 0,
+                Err(_e) => 0,
+            }
+        }
+
+        fn set_decided_idx(&mut self, ld: u64) {
+            let ld_bytes = AsBytes::as_bytes(&ld);
+            self.db.put(b"ld", ld_bytes).unwrap();
+        }
+
+        fn get_accepted_round(&self) -> Ballot {
+            match self.db.get(b"acc_round") {
+                Ok(Some(mut value)) => {
+                    let acc_bytes: &mut [u8] = &mut value;
+                    FromBytes::read_from(acc_bytes).unwrap()
+                }
+                Ok(None) => Ballot::default(),
+                Err(_e) => Ballot::default(),
+            }
+        }
+
+        fn set_accepted_round(&mut self, na: Ballot) {
+            let acc_bytes = AsBytes::as_bytes(&na);
+            self.db.put(b"n_prom", acc_bytes).unwrap();
+        }
+
+        fn set_stopsign(&mut self, s: StopSignEntry) {
+            self.stopsign = Some(s);
+        }
+
+        fn get_stopsign(&self) -> Option<StopSignEntry> {
+            self.stopsign.clone()
+        }
+
+        fn trim(&mut self, trimmed_idx: u64) {
+            //println!("TRIM!");
+            let mut trimmed_log: Vec<T> = self.get_entries(0, self.c_log.next_offset()); // get the entire log, drain it until trimmed_idx
+                                                                                         // println!("the trimmed log before drain {:?}", trimmed_log);
+            trimmed_log.drain(0..trimmed_idx as usize);
+            // println!("the trimmed log after drain {:?}", trimmed_log);
+            let _ = std::fs::remove_dir_all(&self.c_log_path); // remove old log
```

Comment:
Since this is a work around due to some strange stuff in commitlog, can you please add a TODO comment and create an issue later (when this PR is merged) so we don't lose track of this?